tests/test_sim_client.py

```diff
@@ -0,0 +1,310 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asynctest
+import logging
+
+from lsst.ts.hvac.simulator.sim_client import SimClient
+
+logging.basicConfig(format="%(asctime)s:%(levelname)s:%(name)s:%(message)s", level=logging.DEBUG)
+
+
+class SimClientTestCase(asynctest.TestCase):
+    async def setUp(self):
+        self.log = logging.getLogger("SimClientTestCase")
+        self.mqtt_client = SimClient()
+
+    async def test_chiller01P01(self):
+        expected_setpoint_activo = 90
+        expected_comando_encendido = True
+        await self.mqtt_client.chiller01P01(
+            setpoint_activo=expected_setpoint_activo, comando_encendido=expected_comando_encendido
+        )
+        self.assertEqual(expected_setpoint_activo, self.mqtt_client.chiller01_p01.setpointActivo)
+        self.assertEqual(expected_comando_encendido, self.mqtt_client.chiller01_p01.comandoEncendido)
+
+        expected_setpoint_activo = 110
+        with self.assertRaises(ValueError):
+            await self.mqtt_client.chiller01P01(
+                setpoint_activo=expected_setpoint_activo, comando_encendido=expected_comando_encendido
+            )
+
+        expected_setpoint_activo = -11
+        with self.assertRaises(ValueError):
+            await self.mqtt_client.chiller01P01(
+                setpoint_activo=expected_setpoint_activo, comando_encendido=expected_comando_encendido
+            )
+
+    async def test_crack01P02(self):
+        expected_setpoint_humidificador = 90
+        expected_setpoint_deshumidificador = 80
+        expected_set_point_cooling = 70
+        expected_set_point_heating = 60
+        expected_comando_encendido = True
+        await self.mqtt_client.crack01P02(
+            setpoint_humidificador=expected_setpoint_humidificador,
+            setpoint_deshumidificador=expected_setpoint_deshumidificador,
+            set_point_cooling=expected_set_point_cooling,
+            set_point_heating=expected_set_point_heating,
+            comando_encendido=expected_comando_encendido,
+        )
+        self.assertEqual(expected_setpoint_humidificador, self.mqtt_client.crack01_p02.setpointHumidificador)
+        self.assertEqual(
+            expected_setpoint_deshumidificador, self.mqtt_client.crack01_p02.setpointDeshumidificador
+        )
+        self.assertEqual(expected_set_point_cooling, self.mqtt_client.crack01_p02.setPointCooling)
+        self.assertEqual(expected_set_point_heating, self.mqtt_client.crack01_p02.setPointHeating)
+        self.assertEqual(expected_comando_encendido, self.mqtt_client.crack01_p02.comandoEncendido)
+
+        expected_setpoint_humidificador = 110
```

Comment:
Thank you for switching to assertRaises. You might consider `bad_setpoint_humidificador` to emphasis that it should fail. Consider a loop for bad values:
```
for bad_setpoint in (-11, -0.001, 100.001, 110):
    with self.assertRaises(ValueError):
        etc.
```
this will stop on the first error. If you want all of them to run even if one fails then insert a new line `with self.subTest(bad_setpoint=bad_setpoint):` before the `with self.assertRaises` (and indent that code block).

Reply:
Loops can also be handy for valid values. For instance testing that you can turn a system on and then off again.

Reply:
I'll take that into consideration for a future ticket.